src/game.cpp

```diff
@@ -11584,29 +11583,33 @@ void game::chat()
         }
     }
 
-    if (available.empty()) {
-        add_msg(m_info, _("There's no-one close enough to talk to."));
-        if ( !query_yn(_("No one nearby. Yell?")) ) {
-            return;
-        }
-        u.shout();
-
-    } else if (available.size() == 1) {
+    if( available.size() == 1 ) {
```

Comment:
> As for the  available.size() == 1  - it was like this before and currently there is "Y - yell" in NPC menu. The only mechanics change here is the  available.empty()  case - currently (in master) it has a separate y/n prompt, here it allows yell/cancel (for muscle memory and consistency).

Is there any other (standard) way to shout? If so, it's probably better to keep your code: if there is exactly one NPC nearby, talk to that one (after all, the player pressed 'C' to chat, not to shout), otherwise show the menu with the additional shout option.

If there is no other way to shout, always opening the menu is probably better as it allows the player to press a fixed key sequence, 'C' (chat, brings up the menu) and 'a' (shout, always an option), to shout, regardless of any nearby NPCs.